### PR TITLE
Fix TypeScript error line/column

### LINKED ISSUE #97: TSC problem matcher doesn't handle line & column in errors
I have a build at https://github.com/xt0rted/slash-command-action/pull/71 that's failing and the typescript problem matcher is picking up the error, but it's not associating the annotation with the correct line as seen [here](https://github.com/xt0rted/slash-command-action/pull/71/files). This looks to be due to line value also containing the column (`107,11`).

This can be seen using https://regex101.com/ with the values:

```regex
^(?:\s+\d+\>)?([^\s].*)\((\d+|\d+,\d+|\d+,\d+,\d+,\d+)\)\s*:\s+(error|warning|info)\s+(\w{1,2}\d+)\s*:\s*(.*)$
```

```terminal
src/commandHandler.ts(107,11): error TS2322: Type 'Response<ReposGetCollaboratorPermissionLevelResponse>' is not assignable to type 'Response<CollaboratorPermissionLevel>'.
```

### PR DESCRIPTION
**Pull request by Zarel**
Fixes #97

### CODE REVIEW

**Comment by Zarel** [comment]
I tested this on my own repository and it works!

**Comment by Zarel** [comment]
I'm surprised this hasn't gotten any attention after 17 days.

In the meantime, if you want this fixed in your local repository, it's a reasonably simple change:

image

**Comment by Zarel** [comment]
@ericsciple you self-assigned the bug this PR fixes; would you be interested in reviewing this?

**Comment by bryanmacfarlane** [comment]
Sorry.  Been busy on some other changes.  I'll have a look soon.

File: .github/tsc.json
Comment (chancancode) [inline .github/tsc.json:13]: I cannot human-parse this regex, but maybe this should just use the official `$tsc` pattern from vscode? Source: https://github.com/microsoft/vscode/blob/master/extensions/typescript-language-features/package.json#L974-L981 (it is not the same as this one)

Comment (chancancode) [reply .github/tsc.json:13]: I have tested the official regex and can confirm that it works:

```js
// commit this file to .github/tsc.json
{
  "problemMatcher": [
    {
      "owner": "tsc",
      "pattern": [
        {
          "name": "tsc",
          "regexp": "^([^\\s].*)[\\(:](\\d+)[,:](\\d+)(?:\\):\\s+|\\s+-\\s+)(error|warning|info)\\s+TS(\\d+)\\s*:\\s*(.*)$",
          "file": 1,
          "line": 2,
          "column": 3,
          "severity": 4,
          "code": 5,
          "message": 6
        }
      ]
    }
  ]
}
```

```yaml
- name: Set up Node
  uses: actions/setup-node@v1
# Run this _after_ setup-node
- name: Set up tsc problem matcher
  run: echo "::add-matcher::${GITHUB_WORKSPACE}/.github/tsc.json"
```

Comment (Zarel) [reply .github/tsc.json:13]: Here is my regex, free-spaced and syntax highlighted:

image

Here is the official pattern from TypeScript, free-spaced and syntax-highlighted:

image

A TypeScript error message looks like this:

```
lib/net.ts(48,26): error TS1005: '}' expected.
```

Both regexes support that pattern. Mine is modified from the old one in this repository, which supported a variable number of commas.

In "fancy" mode, a TypeScript error message looks like this:

```
lib/net.ts:48:26 - error TS1005: '}' expected.

48  			out += `${key}=${enc odeURIComponent('' + body[key])}`;
    			                     ~~~~~~~~~~~~~~~
```

The official regex also supports this format (although it should never be passed to GitHub Actions in this format, in practice).

Comment (Zarel) [reply .github/tsc.json:13]: In conclusion, the official pattern is probably better.

Do you think the free-spaced annotated version should be put somewhere, for readability?

```
^
# filename
([^\s].*)
# line number in "(1,2)" or ":1:2 -" format
[\(:]
(\d+) [,:] (\d+)
(?:\):\s+|\s+-\s+)
# severity
(error|warning|info)
# error code
\s+ TS(\d+)
# ":"
\s* :
# error message
\s* (.*)
$
```

(It can be compiled into the version in the repository with https://zarel.github.io/regexfree/ )

File: .github/tsc.json
Comment (Zarel) [inline .github/tsc.json:7]:
```suggestion
                    "regexp": "^([^\\s].*)[\\(:](\\d+)[,:](\\d+)(?:\\):\\s+|\\s+-\\s+)(error|warning|info)\\s+TS(\\d+)\\s*:\\s*(.*)$",
```

(Official regex from vscode at https://github.com/microsoft/vscode/blob/cd41dd1f6065db76c77c6e2f659bb50e6f558dec/extensions/typescript-language-features/package.json#L974-L981 )

File: .github/tsc.json
Comment (Zarel) [reply .github/tsc.json]: (One other difference is that the official pattern doesn't consider "TS" part of the error code. There's an argument that it should be kept, for googleability.)

**Review by TimWolla — APPROVED** [review]
This change works for me.

**Comment by TimWolla** [comment]
@bryanmacfarlane Is this one still on your radar?

**Comment by bryanmacfarlane** [comment]
@Zarel any interest in changing the PR to target main instead of master?

**Comment by Zarel** [comment]
@bryanmacfarlane Sure! I just did. Does this mean you're planning to merge this?

**Comment by bryanmacfarlane** [comment]
LGTM.  @ericsciple can you have a quick look?

Also, side question - I think that regex was pulled (or created from) the one that VS Code uses in it's problem matchers.  Anyone know what VS Code is currently using and how this PR compares to it?

**Comment by lumaxis** [comment]
@bryanmacfarlane It's mentioned in one of the comments above, this is the exact same regex that VS Code is currently using: https://github.com/microsoft/vscode/blob/8d9002679558432eca926b6bb4f767eaae6b1068/extensions/typescript-language-features/package.json#L1095-L1106

File: .github/tsc.json
Comment (ericsciple) [inline .github/tsc.json:7]: It might be good to add a unit test that loads the regexp and processes it against a line, asserts expected match groups

I'm worried if this change regresses something, at least we have a baseline unit test suite going forward.

Also i'm curious how this compares to vscode's built-in `tsc` problem matcher or their `tsc-watch` problem matcher. I briefly looked, but wan't able to find those built-in regexps in their repo.

**Comment by bryanmacfarlane** [comment]
@lumaxis are you ok to add unit test(s) per @ericsciple comment?  I think we're good to merge after that.

**Comment by lumaxis** [comment]
@bryanmacfarlane Yea, that sounds like a great idea 👍🏼 Added tests in https://github.com/Zarel/setup-node/pull/1
Either @Zarel can merge my PR directly or @bryanmacfarlane can you merge/cherry-pick them into this branch with maintainer edit powers?

**Comment by Zarel** [comment]
> @bryanmacfarlane Yea, that sounds like a great idea 👍🏼 Added tests in [Zarel#1](https://github.com/Zarel/setup-node/pull/1)
> Either @Zarel can merge my PR directly or @bryanmacfarlane can you merge/cherry-pick them into this branch with maintainer edit powers?

I merged your PR directly. I'm glad to see this is finally getting looked at! :)

File: .github/tsc.json
Comment (bryanmacfarlane) [reply .github/tsc.json:7]: @ericsciple looks like tests are below.  have a quick look

**Comment by bryanmacfarlane** [comment]
LGTM.  any objections @ericsciple ?  (if not, I'll merge tomorrow)

**Review by ericsciple — APPROVED** [review]

**Comment by bryanmacfarlane** [comment]
merged.  Thanks for the contribution!  I'll push the v2 tag forward later today

**Comment by TimWolla** [comment]
> I'll push the v2 tag forward later today

@bryanmacfarlane I believe the tag was not yet updated. At least https://github.com/actions/setup-node/releases still shows a date of December, 16th. Can you check?

**Comment by bryanmacfarlane** [comment]
yep.  doing now :)   Got side tracked on some long live site issues and didn't want to do over the weekend

**Comment by bryanmacfarlane** [comment]
Done. v.2.1.5 and v2 was pushed

https://github.com/actions/setup-node/releases

**Review by Sindhuja365 — APPROVED** [review]

**Pull request merged**
